mass_mailing_custom_unsubscribe/controllers/main.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2015 Antiun Ingeniería S.L. (http://www.antiun.com)
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo.http import request, route
+
+from odoo.addons.mass_mailing.controllers.main import MassMailController
+
+_logger = logging.getLogger(__name__)
+
+
+class CustomUnsubscribe(MassMailController):
+    def reason_form(self, mailing_id, email, res_id, reasons, token):
+        """Get the unsubscription reason form.
+
+        :param mailing.mailing mailing:
+            Mailing where the unsubscription is being processed.
+
+        :param str email:
+            Email to be unsubscribed.
+
+        :param int res_id:
+            ID of the unsubscriber.
+
+        :param str token:
+            Security token for unsubscriptions.
+        """
+        return request.render(
+            "mass_mailing_custom_unsubscribe.reason_form",
+            {
+                "email": email,
+                "mailing_id": mailing_id,
+                "reasons": reasons,
+                "res_id": res_id,
+                "token": token,
+            },
+        )
+
+    @route()
+    def mailing(self, mailing_id, email=None, res_id=None, token="", **post):
+        """Ask/save unsubscription reason."""
+        _logger.debug(
+            "Called `mailing()` with: %r", (mailing_id, email, res_id, token, post)
+        )
+        reasons = request.env["mail.unsubscription.reason"].search([])
+        if not res_id:
+            res_id = "0"
+        res_id = res_id and int(res_id)
```

Comment:
what is the reason for using "0" and not 0?
```suggestion
        res_id = res_id or 0
```
Or better yet, set the default value to zero directly...